### DISCUSSION TITLE
Accessing the source vector from `VectorOperator`

### DISCUSSION THREAD

**Discussion by davidscn** [DavidSCN/mor-coupling#6] [Q&A] [answered]
I don't really get it, but I guess @renefritze knows how to deal with it:

I would like to add deal.II functions in the code handling the coupling, which require a vector reference as input argument `Vector<>&`. However, I don't find a way to fetch it from the `VectorOperator` class.

```python

operator = VectorOperator(DealIIVectorSpace.make_array([dealii.get_solution()]))
dealii.initialize(operator.get_vector_reference)
```

(`get_vector_reference` is just a dummy to illustrate the problem).
Do you have an idea how to do that?

**Comment by renefritze** [comment, marked as answer]
If you take a look at the `VectorOperator` [Docs](https://docs.pymor.org/2021-1-0/autoapi/pymor/operators/constructions/index.html?highlight=vectoroperator#pymor.operators.constructions.VectorOperator)
You can get a vector array from it (`as_source_array()` or range) which then is a `ListVectorArray` with ` DealIIVector`s inside whose `impl` is then what you want I think.

**Reply by davidscn** [reply]
The function `as_sorce_array` throws an exception when using it for `operator`, the function ` as_range_array` returns indeed a `ListVectorArray()`, but I can't get any access to a `DealIIVector`. The `ListVectorArray` is according to python of type `ListVectorArray('??', DealIIVectorSpace(289))`. Is the `DealIIVector` not recognized within this context?

**Reply by renefritze** [reply]
RIght. To get at the `DealIIVector` from the `ListVectorarray` you need to access the `_list` member at the wanted index. [Example](https://github.com/pymor/pymor/blob/64bb8c5c9ec9b4456ae9f508d5651d513827b72d/src/pymor/operators/list.py#L41)

**Reply by davidscn** [reply]
Yup, this works. Auto-complete doesn't show me the `_list` member function.
